src/Collection/Helpers/ConditionParserHelper.php

```diff
@@ -23,11 +24,11 @@
 class ConditionParserHelper
 {
 	/**
-	 * @return array{class-string, string}
+	 * @return array{class-string, string, 2?: mixed}
 	 */
 	public static function parsePropertyOperator(string $condition): array
 	{
-		if (!preg_match('#^(.+?)(!=|<=|>=|=|>|<)?$#', $condition, $matches)) {
+		if (!preg_match('#^(.+?)(!=|<=|>=|=|>|<|~|~start|~end|~both)?$#', $condition, $matches)) {
```

Comment:
Not sure how much it matters here, but ungreedy patterns should be avoided to avoid backtracking, i.e. it should be written as 

```
^(\w++(?:->\w++)*+)(!=|<=|>=|=|>|<|~|~start|~end|~both)?+$
```

Reply:
Fixed here: https://github.com/nextras/orm/commit/3161e291116d327d974eddbe081ef546c8238047